src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,114 @@
+            String nextLine = sc.nextLine();
+            String[] actionArray = nextLine.split("]");
+            String[] actionNextArray = actionArray[1].split(":");
+            String action = actionNextArray[0].strip();
+            switch(action) {
+                case "Todo": {
+                    String[] taskArray = nextLine.split(":");
+                    String task = taskArray[1].strip();
+                    Task savedTask = new Todo(task);
+                    if (nextLine.substring(1, 2).equals("X")) {
+                        savedTask.check();
+                    }
+                    tasksList.add(savedTask);
+                    break;
+                }
+                case "Deadline": {
+                    String[] frontTaskArray = nextLine.split(":");
+                    String[] backTaskArray = frontTaskArray[1].split("\\[");
+                    String task = backTaskArray[0].strip();
+                    String[] frontByArray = nextLine.split("\\[");
+                    String[] backByArray = frontByArray[1].split("]");
+                    String by = backByArray[0].strip();
+                    LocalDate storageBy = LocalDate.parse(by, DateTimeFormatter.ofPattern("MMM dd yyyy"));
+                    String outputBy = storageBy.format(DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+                    Task savedTask = new Deadline(task, outputBy);
+                    if (nextLine.substring(1, 2).equals("X")) {
+                        savedTask.check();
+                    }
+                    tasksList.add(savedTask);
+                    break;
+                }
+                case "Event": {
+                    String[] frontTaskArray = nextLine.split(":");
+                    String[] backTaskArray = frontTaskArray[1].split("\\[");
+                    String task = backTaskArray[0].strip();
+                    String[] frontFromArray = nextLine.substring(3).split("\\[");
+                    String[] backFromArray = frontFromArray[1].split("-");
+                    String from = backFromArray[0].strip();
+                    String[] frontToArray = nextLine.split("-");
+                    String[] backToArray = frontToArray[1].split("]");
+                    String to = backToArray[0].strip();
+                    LocalDate storageFrom = LocalDate.parse(from, DateTimeFormatter.ofPattern("MMM dd yyyy"));
+                    String outputFrom = storageFrom.format(DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+                    LocalDate storageTo = LocalDate.parse(to, DateTimeFormatter.ofPattern("MMM dd yyyy"));
+                    String outputTo = storageTo.format(DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+                    Task savedTask = new Event(task, outputFrom, outputTo);
+                    if (nextLine.substring(1, 2).equals("X")) {
+                        savedTask.check();
+                    }
+                    tasksList.add(savedTask);
+                    break;
```

Comment:
This one can apply SLAP (Single Level of Abstraction Principle) and change to a method, especially since there are similar code used in other case blocks too.